react/providers/api/provider.tsx

```diff
@@ -1,30 +1,39 @@
-import React, { useEffect, useReducer } from 'react'
-import { useChains } from '../../hooks/mod.ts'
-import { APIContext } from './context.ts'
-import { apiProvidersReducer } from './reducer.ts'
-import { useConfig } from '../../mod.ts'
+import React, { useEffect, useReducer } from "react";
+import { useChains } from "../../hooks/mod.ts";
+import { APIContext } from "./context.ts";
+import { apiProvidersReducer } from "./reducer.ts";
+import { useConfig } from "../../mod.ts";
+import { Sc, ScProvider } from "../../../core/mod.js";
 import { ApiPromise, WsProvider } from "../../../core/mod.ts";
 
 export const APIProvider: React.FC<React.PropsWithChildren<any>> = ({
   children,
 }) => {
-  const chains = useChains()
-  const { chainRpcs } = useConfig()
-  const [apis, dispatch] = useReducer(apiProvidersReducer, {})
+  const chains = useChains();
+  const { chainRpcs, api } = useConfig();
+  const [apis, dispatch] = useReducer(apiProvidersReducer, {});
 
   useEffect(() => {
     chains.forEach((chain) => {
-      const provider = new WsProvider(chainRpcs[chain.id] || chain.rpcs[0])
+      const provider = new WsProvider(chainRpcs[chain.id] || chain.rpcs[0]);
 
       ApiPromise.create({ provider }).then((api) => {
         dispatch({
-          type: 'ADD_API_PROVIDER',
+          type: "ADD_API_PROVIDER",
           chainId: chain.id,
           apiProvider: { api, provider },
-        })
-      })
-    })
-  }, [chains, chainRpcs])
+        });
+      });
+    });
+  }, [chains, chainRpcs]);
 
-  return <APIContext.Provider value={{ apis }} children={children} />
-}
+  useEffect(() => {
+    if (api?.default === "light-client") {
+      const provider = new ScProvider(Sc, "all");
+
+      provider.connect();
+    }
+  }, [api]);
```

Comment:
@DoubleOTheven in [this](https://github.com/paritytech/useink/pull/76#discussion_r1212029607) comment you mentioned

> Then inside of ApiProvider you use useConfig() to get the values and create the instances accordingly. This will update automatically when the config is changed


I tried to use `useEffect` to update automatically, but not entirely sure if this is the right approach and/ or if the other `useEffect` becomes obsolete.

Let me know how do you envision `APIProvider` to behave.

Reply:
If you think there is a better approach you can suggest it. I was only giving some ideas.

`useEffect` can be used to update the state of ApiProvider if some other state changes.